test/models/concerns/host_managed_extensions_test.rb

```diff
@@ -43,6 +43,16 @@ def test_full_text_search
       refute_includes found, other_host
     end
 
+    def test_host_status_reset
+      host = FactoryBot.create(:host)
+      host.host_statuses.delete_all
+      Katello::SubscriptionStatus.create!(host: host, :status => Katello::SubscriptionStatus::VALID)
+      host.reload
+      assert_equal Katello::SubscriptionStatus::VALID, host.host_statuses.first.status
+      host.reset_katello_status
```

Comment:
if you set build to true on the host and save rather than calling reset_katello_status then we can test the full behavior

Reply:
Hmm I didnt want to deal with the host "queue" in my test case. Will play with it though.

Reply:
If it's a pain then don't worry about it

Reply:
Yeh tried simpler version but its a pain because I dont think the queue runs in a test env when we save?